README.md
www.conductorproductions.com
============================

[![Build Status](https://travis-ci.org/remomueller/www.conductorproductions.com.svg)](https://travis-ci.org/remomueller/www.conductorproductions.com)
[![Dependency Status](https://gemnasium.com/remomueller/www.conductorproductions.com.svg)](https://gemnasium.com/remomueller/www.conductorproductions.com)
[![Code Climate](https://codeclimate.com/github/remomueller/www.conductorproductions.com/badges/gpa.svg)](https://codeclimate.com/github/remomueller/www.conductorproductions.com)


CHANGELOG.md
## 10.0.0

### Enhancements
- **General Changes**
  - Updated user interface across site
  - Added new landing page
  - Added new services page
  - Added new creators page
  - Updated administrative dashboard
- **Gem Changes**
  - Updated to ruby 2.5.0-rc1
  - Updated to rails 5.1 defaults
  - Updated to rails 5.2.0.beta2
  - Updated to bootstrap 4.0.0.beta2
  - Updated to carrierwave 1.2.1
  - Updated to haml 5.0.4
  - Updated to kaminari 1.1.1

### Tests
- Added tests to assure user passwords can be reset

## 0.9.3 (December 20, 2017)

### Enhancement
- **General Changes**
  - Tom Brady link now opens in another windo

## 0.9.2 (December 20, 2017)

### Enhancement
- **General Changes**
  - Updated copy on A Year in Review 2017 page

## 0.9.1 (December 19, 2017)

### Enhancement
- **General Changes**
  - Added A Year in Review 2017 page

## 0.9.0 (October 21, 2017)

### Enhancements
- **General Changes**
  - Login cookies are now cross subdomain and work between www and non-www URLs
  - The DRTV page now redirects to the Work page
- **Member Changes**
  - Admins can now create and edit director and other team member pages
- **Video Changes**
  - Admins can now set Client Name and Agency Name for videos
  - Hovering over videos now displays Client Name and Agency Name
- **Gem Changes**
  - Updated to rails 5.1.4
  - Updated to pg 0.21.0
  - Updated to devise 4.3.0
  - Updated to carrierwave 1.1.0
  - Updated to haml 5.0.3
  - Updated to simplecov 0.15.1

## 0.8.2 (October 21, 2017)

### Enhancements
- **Gem Changes**
  - Updated to rails 5.0.6

## 0.8.1 (October 19, 2017)

### Bug Fix
- Fixed a bug that prevented passwords from being reset

## 0.8.0 (April 22, 2017)

### Enhancements
- **General Changes**
  - Pages redesigned by PixelCure
    - Home Page
    - Our Work
    - Directors
    - Clients
    - Contact
- **Video Changes**
  - System admins can now configure videos on the home page
- **Gem Changes**
  - Updated to ruby 2.4.1
  - Updated to rails 5.0.2
  - Updated to jquery-rails 4.3.1
  - Updated to carrierwave 1.0.0
  - Updated to bootstrap-sass 3.3.7
  - Updated to jquery-ui-rails 6.0.1
  - Updated to kaminari 1.0.1
  - Updated to simplecov 0.14.1
  - Updated to pg 0.20.0
  - Added autoprefixer-rails

### Bug Fix
- Fixed a bug that prevented gallery photos from being rearranged after
  uploading a new photo

## 0.7.1 (February 6, 2017)

### Enhancements
- **General Changes**
  - Updated the link to the first video on the landing page

## 0.7.0 (December 20, 2016)

### Enhancements
- **General Changes**
  - Added a new Year in Review page

### Bug Fix
- Fixed a JavaScript that occurred when adding event listener to non-existent
  video

## 0.6.0 (October 16, 2016)

### Enhancements
- **General Changes**
  - Added new video to landing page
  - Added dropdown menu for logged in users
- **Email Changes**
  - Updated styling of emails
- **Client Changes**
  - Top level menu items are now hidden if they are empty
  - Subcategories no longer display in side menu if the containing gallery only
    has one subcategory
- **Document Changes**
  - Documents can now upload a PDF and Word version side-by-side
  - Categories with only one document no longer show latest and archived submenu
- **Gallery Changes**
  - Locations have been renamed to galleries
  - Gallery and photo positions can be reordered
- **Project Changes**
  - Top level REFERENCES has been added
  - Default categories for projects have been updated
  - Project usernames can no longer contain spaces
  - Admin and client logins have been combined
  - If a category has a single location, and no associated documents or embeds,
    then the client is redirected to the single location gallery view
  - Simplified links on project index to highlight preview link
  - URL slugs for projects, categories, and galleries now automatically fill in
    when entering a new name
  - Documents, embeds, and galleries can now be archived on from their index
    pages
  - Project main page layout has been updated while editing project
- **Gem Changes**
  - Updated to ruby 2.3.1
  - Updated to rails 5.0.0.1
  - Updated to coffee-rails 4.2
  - Updated to jbuilder 2.5
  - Updated to jquery-rails 4.2.1
  - Updated to turbolinks 5
  - Updated to devise 4.2.0
  - Updated to simplecov 0.12.0
  - Updated to kaminari 0.17.0
  - Updated to haml 4.0.7
  - Updated to carrierwave 0.11.2
  - Updated to pg 0.19.0

### Refactoring
- Removed unused references to tumblr API
- Cleaned up routes file
- Cleaned up inviting and adding users to project
- Updated all views to use haml

### Bug Fixes
- Fixed a bug that prevented a project username from being reused if a deleted
  project had the same username
- Fixed a bug that prevented videos from hiding correctly on Work and DRTV pages
  when a video thumbnail was clicked

## 0.5.1 (April 23, 2016)

### Enhancements
- **Client Changes**
  - Internal client URLs are now friendly forwarded
  - Project submenus are now visible by default
- **Project Changes**
  - Project owners and editors can now archive projects
- **Email Changes**
  - Contacts through the website now go to a new email address
- **General Changes**
  - Added a JSON API for the site version
- **Gem Changes**
  - Updated to ruby 2.3.0
  - Updated to rails 4.2.6
  - Updated to pg 0.18.4
  - Updated to devise 4.0.0
  - Updated to jquery-rails 4.1.1
  - Updated to carrierwave 0.11.0
  - Removed minitest-reporters
  - Added simplecov configuration file

### Bug Fix
- Deleted users are no longer able to log in
- Fixed an issue with duplicate scrollbars on some pages

## 0.5.0 (August 1, 2015)

### Enhancements
- **Client Changes**
  - Updated the internal client pages with cleaner layout
  - Client menu sidebar updated to match root menu design
  - Small footer graphic on client login page has been fixed
  - All latest and archived category documents and embeds are now listed in the
    archive menu in the sidebar

gems.rb
# frozen_string_literal: true

source "https://rubygems.org"

gem "rails",                "5.2.0.beta2"

# Database Adapter
gem "pg",                   "0.21.0"

# Gems used by project
gem "autoprefixer-rails"
gem "bcrypt",               "~> 3.1.10"
gem "bootstrap",            "~> 4.0.0.beta2"
gem "carrierwave",          "~> 1.2.1"
# gem "devise",               "~> 4.3.0"
gem "devise", git: "https://github.com/plataformatec/devise", ref: "4f2e915"
gem "figaro",               "~> 1.1.1"
gem "font-awesome-rails",   "~> 4.7.0"
gem "haml",                 "~> 5.0.4"
gem "jquery-ui-rails",      "~> 6.0.1"
gem "kaminari",             "~> 1.1.1"
gem "mini_magick"

# Rails Defaults
gem "coffee-rails",         "~> 4.2"
gem "sass-rails",           "~> 5.0"
gem "uglifier",             ">= 1.3.0"

gem "jbuilder",             "~> 2.5"
gem "jquery-rails",         "~> 4.3.1"
gem "turbolinks",           "~> 5"

# Testing
group :test do
  gem "minitest"
  gem "rails-controller-testing"
  gem "simplecov", "~> 0.15.1", require: false
end
